File: 2025-1/sesion-de-clase-19/Cliente.cpp
/* 
 * File:   Cliente.cpp
 * Author: erichuiza
 * 
 * Created on May 30, 2025, 11:56 AM
 */

#include "Cliente.h"


Cliente::Cliente(
    int dni, 
    const string nombre, 
    const string telefono, 
    const string correo
) : Persona(dni, nombre) {
    this->setTelefono(telefono);
    this->setCorreo(correo);
}

string Cliente::getCorreo() const {
    return correo;
}

void Cliente::setCorreo(const string correo) {
    this->correo = correo;
}

string Cliente::getTelefono() const {
    return telefono;
}

void Cliente::setTelefono(const string telefono) {
    this->telefono = telefono;
}

bool Cliente::cargarDatos(ifstream& archivo) {
    if (Persona::cargarDatos(archivo)) {
        getline(archivo, telefono, ',');
        getline(archivo, correo, '\n');
        
        this->setTelefono(telefono);
        this->setCorreo(correo);
        
        return true;
    }
    return false;
}

void Cliente::imprimir(ostream& os) const {
    Persona::imprimir(os);
    os 
        << setw(20) << this->telefono
        << setw(20) << this->correo
        << endl;
}

File: 2025-1/sesion-de-clase-24-1/src/Cliente.java

import java.util.Scanner;

/**
 *
 * @author erichuiza
 */
public class Cliente extends Persona {
    private String telefono;
    private String correo;
    
    public Cliente() {
    }
    
    public Cliente(int dni, String nombre, String telefono, 
            String correo) {
        super(dni, nombre);
        this.telefono = telefono;
        this.correo = correo;
    }
    
    public String getTelefono() {
        return telefono;
    }

    public void setTelefono(String telefono) {
        this.telefono = telefono;
    }

    public String getCorreo() {
        return correo;
    }

    public void setCorreo(String correo) {
        this.correo = correo;
    }
    
    @Override
    protected void cargarDatos(Scanner scanner) {
        super.cargarDatos(scanner);
        this.telefono = scanner.next();
        this.correo = scanner.next();
    }

    @Override
    protected void imprimir() {
        super.imprimir();
        System.out.println("Telefono: " + this.telefono);
        System.out.println("Correo: " + this.correo);
    }
}
